components/Pagination/PaginationItem/PaginationItem.js

```diff
@@ -9,16 +10,21 @@ PaginationItem.propTypes = {
   children: node.isRequired,
   isCurrent: bool,
   pathname: string.isRequired,
+  query: object,
   testId: string.isRequired,
   value: number,
 };
 
 PaginationItem.defaultProps = {
   isCurrent: false,
+  query: {},
   value: undefined,
 };
 
-function PaginationItem({ children, isCurrent, pathname, testId, value }) {
+function PaginationItem({ children, isCurrent, pathname, query, testId, value }) {
+  const relevantQueryStringObject = omit(query, ['page']);
+  const realURL = { pathname: pathname.replace('[page]', value), query: relevantQueryStringObject };
```

Comment:
actually a little confused about this the more I look at it. the pathname usually has the dynamic route [page] in it- why would you need to omit this from the query? why would it be in the query? the query should be something like `{ page: 1, q: _userQuery_}`

Reply:
looked up `omit`, saw that it is just an array of paths, and that you are literally omitting the `page` param from the query.. still don't quite understand why though

Reply:
Are you talking about the value for `pathname` or `query`.

### For query...
A strange caveat is that the route param for dynamic routes gets parsed as in the query object. If we _didn't_ then `?page=x` would always show up in the query string :(

### For pathname...
Just following documentation on this one! https://nextjs.org/docs/api-reference/next/link#dynamic-routes